complete.py

```diff
@@ -0,0 +1,795 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+#####Imports#####
+#https://github.com/dominicrufa/coddiwomple for coddiwomple imports
+from coddiwomple.openmm.integrators import OMMLI #OpenMMLangevinIntegrator
+from coddiwomple.openmm.propagators import OMMBIP #OpenMMBaseIntegratorPropagator
+from coddiwomple.openmm.reporters import OpenMMReporter #trajectory reporter object
+from openmmtools.states import ThermodynamicState, SamplerState
```

Comment:
I am not really familiar with openmmtools -- just to clarify:
- you are using ThermoDynamicState as a way to compute the reduced_potential with respect to the ensemble?
- you are using SamplerState as a way to index/slice the topology and coordinates?

Reply:
i use `ThermodynamicState.reduced_potential` method on a sampler state as a convention to compute the reduced potential; the sampler state has to be updated from the context to update the box vectors and potential energy to the sampler state

Reply:
sampler state holds positions, velcoities, and box vectors

Reply:
thanks for clarifying!